tests/test_batch_run.py

```diff
@@ -47,6 +47,7 @@ def __init__(
         variable_agent_param=None,
         fixed_model_param=None,
         schedule=None,
+        agent_reporters=True,
```

Comment:
This variable is a boolean, and then the type changes into a dictionary, which is actually used for a different purpose. This is confusing. It's best to rename this boolean variable to `enable_agent_reporters` or something.